app/Services/Rogue.php

```diff
@@ -99,4 +97,23 @@ public function updatePost($postId, $input)
 
         return $post;
     }
+
+    /**
+     * Get an action id from Rogue based on CallPower campaign id.
+     *
+     * @param int $callpowerCampaignId
+     * @return array $action
+     */
+    public function getActionIdFromCallPowerCampaignId($callpowerCampaignId)
+    {
+        $action = $this->asClient()->get('v3/actions', [
+            'filter' => ['callpower_campaign_id' => $callpowerCampaignId],
+        ]);
+
+        if (! $action['data']) {
```

Comment:
This should be checking if there's an item at index `['data'][0]` or `['data'][0]['id']` right? I believe we'll have a `data` index even if there isn't a matching CallPower campaign… it'll just be empty.

Reply:
you're totally right, thank you for spotting that! Updated!